trace_context/HTTP_HEADER_FORMAT.md

```diff
@@ -0,0 +1,159 @@
+ is used as a delimiter between fields.
+
+#### Trace-id
+
+Is the ID of the whole trace forest. It is represented as a 16-bytes array, e.g., 
+`4bf92f3577b34da6a3ce929d0e0e4736`. All bytes 0 is considered invalid.
+
+Implementation may decide to completely ignore the traceparent if the trace-id is invalid.
+
+#### Span-id
+
+Is the ID of the caller span (parent). It is represented as a 8-bytes array, e.g., 
+`00f067aa0ba902b7`. All bytes 0 is considered invalid.
+
+Implementation may decide to completely ignore the traceparent if the span-id is invalid.
+
+#### Trace-options
+
+Controls tracing options such as sampling, trace level etc. It is a 1-byte representing a 8-bit 
+unsigned integer. The flags are recommendations given by the caller rather than strict rules to 
+follow for 3 reasons:
+
+1. Trust and abuse.
+2. Bug in caller
+3. Different load between caller service and callee service might force callee to down sample.    
+
+##### Bits behavior definition (01234567):
+* The least significant bit (the 7th bit) provides recommendation whether the request should be 
+traced or not (`1` recommends the request should be traced, `0` means the caller does not
+make a decision to trace and the decision might be deferred). When `trace-options` is missing
+the default value for this bit is `0`
+* The behavior of other bits is currently undefined.
+
+#### Examples of HTTP headers
+
+*Valid sampled traceparent:*
+
+```
+Value = 00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01
+base16(<Version>) = 00
+base16(<TraceId>) = 4bf92f3577b34da6a3ce929d0e0e4736
+base16(<SpanId>) = 00f067aa0ba902b7
+base16(<TraceOptions>) = 01  // sampled
+```
+
+*Valid not-sampled traceparent:*
+
+```
+Value = 00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-00
+base16(<Version>) = 00
+base16(<TraceId>) = 4bf92f3577b34da6a3ce929d0e0e4736
+base16(<SpanId>) = 00f067aa0ba902b7
+base16(<TraceOptions>) = 00  // not-sampled
+```
+
+# Trace-State Field
+
+## Header name
+
+`tracestate`
+
+As multiple trace graphs can be present, multiple trace state headers are
+allowed. Values can be combined in a single header according to the [RFC 7230](https://tools.ietf.org/html/rfc7230#page-24).
+
+## Header value
+
+`name1[=value1[;properties1]],name2[=value2[;properties2]]`
+
+**Limits:**
+Maximum length of a combined header MUST be less than 512 bytes. 
+
+## Name format
+
+Url encoded string. Spaces are allows before and after the name. Header with the trimmed name and with spaces before and after name MUST be considered identical.
+
+Names `id`, `span-id`, `trace-id`, `sampled` are reserved. These properties are defined in `traceparent` header.
+
+## Value format
+
+Value starts after equal sign and ends with the special character `;`, separator `,` or end of string. Value represents a url encoded string and case sensitive. Spaces are allowed in the beginning and the end of the value. Value with spaces before and after MUST be considered identical to the trimmed value. 
```

Comment:
Does most people agree about this format of state.

Reply:
What's the purpose or benefit of including semicolon `;` as a value-ending character? Can we (or should we) just ignore it and say that commas `,` and end-of-string are the only things that end a value?

I guess I'm assuming that "value" in this context means the entire opaque vendor-specific trace state, and since it's opaque they could use `;` if they want to but by itself it has no special meaning and should therefore not be called out - is that the case or am I misunderstanding?

Reply:
It's also implicit in here that the value-ending characters (e.g. `,`) are therefore reserved and should not be used in a vendor's opaque trace state value, but might be worth explicitly calling it out.

Reply:
done (the old formatting was unnecessary in opaque, but restricted characters are)